tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -4274,3 +4559,59 @@ def from_learner(cls, learner: AbstractTabularLearner):
         predictor = cls(label=learner.label, path=learner.path)
         predictor._set_post_fit_vars(learner=learner)
         return predictor
+
+
+def _sub_fit(self, train_data, time_limit, ds_fit_kwargs, ag_fit_kwargs, ag_post_fix_kwargs, custom_val_data=None):
+    holdout_frac = ds_fit_kwargs.get("holdout_frac", None)
+    train_indices = ds_fit_kwargs.get("train_indices", None)
+    if holdout_frac is not None:
+        train_data, val_data = generate_train_test_split_combined(
+            train_data,
+            label=self.label,
+            problem_type=self.problem_type,
+            test_size=holdout_frac,
+            random_state=42,
+        )
+    elif train_indices is not None:
+        val_indices = ds_fit_kwargs.get("val_indices", None)
+        val_data = train_data.iloc[val_indices]
+        train_data = train_data.iloc[train_indices]
+    elif custom_val_data is not None:
+        train_data = train_data
+        val_data = custom_val_data
+    else:
+        raise ValueError("Unsupported validation procedure during dynamic stacking!")
+
+    set_logger_verbosity(ag_fit_kwargs["verbosity"])
+    org_learner = copy.deepcopy(self._learner)
```

Comment:
Why are we creating a copy of the learner instead of a copy of the predictor? (both are probably fine, but wanted to know the reasoning)

Reply:
To my understanding the only part of the predictor that is affected / needed for a reset to the original state before a fit is the learner. I think copying the predictor would be more expensive (copying useless stuff) than copying the learner. But for safety, we could also copy the predictor and reset the object this way.